python_modules/automation/automation/eval/cli.py

```
@@ -0,0 +1,330 @@
+    if not config_path.exists():
+        raise click.UsageError(f"Configuration file {config_path} not found")
+
+    with open(config_path) as f:
+        config_data = yaml.safe_load(f)
+
+    return EvalConfig(**config_data)
+
+
+def load_sessions(eval_dir: Path) -> dict[str, dict[str, Any]]:
+    """Load all session files from the directory."""
+    sessions = {}
+    session_files = list(eval_dir.glob("*.json"))
+
+    # Exclude cache files
+    session_files = [f for f in session_files if not f.name.startswith("metric-")]
+
+    for session_file in session_files:
+        session_id = session_file.stem
+        try:
+            with open(session_file) as f:
+                session_data = json.load(f)
+
+            # Validate required fields
+            if not isinstance(session_data, dict):
+                click.echo(f"Warning: Skipping {session_file.name} - not a valid JSON object")
+                continue
+
+            if (
+                "input" not in session_data
+                or "output" not in session_data
+                or "timestamp" not in session_data
+            ):
+                click.echo(f"Warning: Skipping {session_file.name} - missing required fields")
+                continue
+
+            sessions[session_id] = session_data
+        except json.JSONDecodeError:
+            click.echo(f"Warning: Skipping {session_file.name} - invalid JSON")
+            continue
+
+    if not sessions:
+        raise click.UsageError("No valid session files found")
+
+    return sessions
+
+
+def load_results(eval_dir: Path, metric: Metric) -> dict[str, dict[str, Any]]:
+    """Load cached results for a metric."""
+    cache_file = eval_dir / f"metric-{metric.get_hash()}.json"
+
+    if not cache_file.exists():
+        return {}
+
+    try:
+        with open(cache_file) as f:
+            return json.load(f)
+    except (OSError, json.JSONDecodeError):
+        # Treat corrupted cache as empty
+        return {}
```

would prefer to fail hard i think